python/cudf/cudf/dataframe/series.py

```diff
@@ -86,12 +86,16 @@ def __init__(
         elif isinstance(data, Index):
             name = data.name
             data = data.as_column()
+            if dtype is not None:
```

Comment:
@galipremsagar if you've changed `as_column` to use the dtype argument on Indexes, could you use it that way here? If dtype=None, nothing will change. if dtype is not none, you'll typecast.

Reply:
Chatted offline. We probably want this behavior to be:

1) Default to the existing dtype of an object, if it has one
2) Optionally override that default based on the keyword argument

@shwina may want to weigh in if he has a different thinking around type handling